app/classifier/classifier.jsx

```diff
@@ -78,40 +75,23 @@ class Classifier extends React.Component {
     }
 
     if (nextProps.classification !== this.props.classification) {
-      const { annotations } = nextProps.classification;
+      const annotations = nextProps.classification.annotations.slice();
       this.setState({ annotations });
     }
   }
 
   componentDidUpdate(prevProps, prevState) {
+    const currentAnnotation = this.state.annotations[this.state.annotations.length - 1];
+    const prevAnnotation = prevState.annotations[prevState.annotations.length - 1];
+
     if (prevProps.classification !== this.props.classification) {
       prevProps.classification.stopListening('change', this.updateAnnotations);
       this.props.classification.listen('change', this.updateAnnotations);
       this.updateAnnotations();
     }
 
-    // We need to know when the active task changes in order to open the
-    // feedback modal. Unfortunately, since we mutate props, we can't compare
-    // state very easily. So we store the current task in state, check to see
-    // when it changes, update state and use _that_ change to check the
-    // _previous_ task for feedback.
-    //
-    // The second part of this hack is using forceUpdate - there's a re-render
-    // happening simultaneously somewhere, so this ensures the subsequent saved
-    // of the new task id to state only happens once, since it's not in the same
-    // batched update.
-    //
-    // This is a really hacky solution, but fixing this properly would required
-    // an extensive rewrite :( Note also that using setState in
-    // componentDidUpdate is a linting error, and with good reason.
-    const currentTaskId = _.get(_.last(this.state.annotations), 'task', '');
-
-    if (prevState.currentTaskId !== currentTaskId) {
-      this.setState({ currentTaskId }, () => this.forceUpdate());
-    }
-
-    if (prevState.currentTaskId !== this.state.currentTaskId) {
-      this.checkForFeedback(prevState.currentTaskId);
+    if (prevAnnotation && currentAnnotation.task !== prevAnnotation.task) {
```

Comment:
I'm not seeing feedback pop-up anymore, I think because it's not getting past the conditional comparing `currentAnnotation.task !== prevAnnotation.task`

Reply:
My fault, I think. It was working until I rebased.

Reply:
Looking at this now. The Done button at the end of a workflow doesn't change the current task, so feedback can't be triggered for the last task in a workflow.

Reply:
This has now confused me because @rogerhutchings original hack shouldn't work either, since it also triggered feedback when the task changes.

Reply:
I broke the rules lookup when I removed `_.get()`. The correct lookup is `feedback.rules[taskKey]`. Fixed now.

Reply:
That said, I'm still confused as to how the feedback check is triggered at the end of a classification.

Reply:
My bad, this wasn't the right line of code to refer to, as noted by your fix with `feedback.rules[taskKey]`, the feedback on single task workflows is triggered through `completeClassification` which is sent through to https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/classifier/task-nav.jsx#L92.

Reply:
Ah yes, I missed that `this.checkFeedback` is called in more than one place.

It would make sense to have a callback, from the task navigation, for task changes too. That might be more reliable than listening to annotation changes.